DeleteUserDataActivity/backupAndDelete.ts

```diff
@@ -48,60 +40,59 @@ import {
  * @param iterator an iterator of every result from the db
  */
 const executeRecursiveBackupAndDelete = <T>(
-  deleteSingle: (item: T) => Promise<Either<QueryError, string>>,
+  deleteSingle: (item: T) => TaskEither<CosmosErrors, string>,
   userDataBackup: IBlobServiceInfo,
   makeBackupBlobName: (item: T) => string,
-  iterator: IResultIterator<T>
+  iterator: AsyncIterator<ReadonlyArray<Either<Errors, T>>>
 ): TaskEither<
   // tslint:disable-next-line: use-type-alias
   DataFailure,
   readonly T[]
 > =>
-  tryCatch(iterator.executeNext, toError)
+  tryCatch(iterator.next, toError)
     // this is just type lifting
-    .foldTaskEither<DataFailure, Option<readonly T[]>>(
+    // tslint:disable-next-line: readonly-array
+    .foldTaskEither<DataFailure, readonly T[]>(
       e => fromLeft(toQueryFailure(e)),
-      e => fromEither(e).mapLeft(toQueryFailure)
+      e =>
```

Comment:
can this be done using the new [filterAsyncIterator](https://github.com/pagopa/io-functions-commons/pull/82/files#diff-6da8d5eb573fbd4e1a43e00a4d44a32dR52) ?

Reply:
Yes, once we publish new `io-functions-commons` version

Reply:
Maybe if we use `filterAsyncIterator` on top of tasks chain, we cannot manage and return an error if  at least an element of result array was left. What do you think @gunzip ?